### PR TITLE
Update TF official devel images to Ubuntu 20.04 and precommit hooks

### PR DESCRIPTION
**Pull request by bhack**
Update TF official devel images to Ubuntu 20.04 LTS and install pytlint.

`ci_build.sh` is currently broker for `ci_sanity.sh` caused by a misalignment  between CI sanity and local images on developer machine (See https://github.com/tensorflow/tensorflow/issues/47989).
Now that https://github.com/tensorflow/tensorflow/pull/48291 is merged this will let developer/contributors to locally execute python linting steps in the docker devel container with:
```
docker run --rm -it -v $PWD:/tensorflow -w /tensorflow tensorflow/tensorflow:devel tensorflow/tools/ci_build/ci_sanity.sh --pylint --incremental
```

It could be nice if later the full CI sanity could be run exactly on the same image where developer are working every day to avoid other potential misalignment.

If this works we could change the documentation for linting. 

I could expand this PR or with a new one to introduce other steps (e.g. include `--clang_format` ?) so that we could have a working minimal configuration that could be also contributed in the repository as a  pre-commit hook  considering that the the whole `ci_sanity.sh` is too heavy to be run as a git hook.

/cc @mihaimaruseac @angerson

### CODE REVIEW

**Comment by kuraga** [comment]
:+1:

**Comment by bhack** [comment]
> :+1: It would be great to solve #29232, too.

That ticket Is merged. Are you sure is it correct?

**Comment by kuraga** [comment]
> > +1 It would be great to solve #29232, too.
> 
> That ticket Is merged. Are you sure is it correct?

Sorry, updated.

**Comment by bhack** [comment]
For the other images we could probably waiting for the Dockerfile "jungle" refactoring at https://github.com/tensorflow/build/pull/21

**Comment by angerson** [comment]
Thanks for this -- how does this do on the tests?

**Comment by bhack** [comment]
> Thanks for this -- how does this do on the tests?

Do we have a string to execute checkout tests against a pip installed TF? Or do you want to execute tests after compilation in the container?

**Comment by bhack** [comment]
> asm_images --release nightly --arg _TAG_PREFIX="" --build_images --run_tests_path=$(realpath tests)

Yes I know this command it is in the Readme but that tests require `build_cpu.sh` and so we are in a similar case as https://github.com/tensorflow/tensorflow/pull/48421. I need hours and hours of compile time to test every image on a local machine.

**Comment by bhack** [comment]
I've added a pre-commit hook prototype.

File: tensorflow/tools/dockerfiles/dockerfiles/arm64v8/devel-cpu-arm64v8.Dockerfile [hunk 1]
```diff
@@ -23,8 +23,11 @@ ARG UBUNTU_VERSION=18.04
 
 FROM ubuntu:${UBUNTU_VERSION} AS base
 
+ARG DEBIAN_FRONTEND="noninteractive"
+
 RUN apt-get update && apt-get install -y --no-install-recommends \
         build-essential \
+        clang-format \
```
Comment (bhack) [inline tensorflow/tools/dockerfiles/dockerfiles/arm64v8/devel-cpu-arm64v8.Dockerfile:30]: With "clang-format-10` that is available in this Ubuntu 20.04 upgrade we could use `clang-format-10 --dry-run --Werror` if we don't like the automatic pre-commit formatting.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+   -   id: pylint
+       name: pylint
+       entry: pylint
+       language: system
+       types: [python]
+ - repo: local
+   hooks:
+   -   id: clang-format
+       name: clang-format
+       entry: clang-format -i
```
Comment (bhack) [inline .pre-commit-config.yaml:13]: With "clang-format-10` that is available in this Ubuntu 20.04 upgrade we could use `clang-format-10 --dry-run --Werror` if we don't like the automatic pre-commit formatting.

File: tensorflow/tools/dockerfiles/dockerfiles/devel-cpu.Dockerfile [hunk 3]
```diff
@@ -23,8 +23,11 @@ ARG UBUNTU_VERSION=18.04
 
 FROM ubuntu:${UBUNTU_VERSION} AS base
 
+ARG DEBIAN_FRONTEND="noninteractive"
+
 RUN apt-get update && apt-get install -y --no-install-recommends \
         build-essential \
+        clang-format \
```
Comment (bhack) [inline tensorflow/tools/dockerfiles/dockerfiles/devel-cpu.Dockerfile:30]: We could install `clang-format-10` if we prefer `dry-run` but we always need to sync version with the CI Sanity VM

File: tensorflow/tools/dockerfiles/dockerfiles/devel-gpu.Dockerfile [hunk 4]
```diff
@@ -38,6 +38,7 @@ ARG LIBNVINFER_MAJOR_VERSION=7
 SHELL ["/bin/bash", "-c"]
 RUN apt-get update && apt-get install -y --no-install-recommends \
         build-essential \
+        clang-format \
```
Comment (bhack) [inline tensorflow/tools/dockerfiles/dockerfiles/devel-gpu.Dockerfile:41]: We could install `clang-format-10` if we prefer `dry-run` but we always need to sync version with the CI Sanity VM

**Comment by bhack** [comment]
/cc @angerson How we could public test that this update will  lint with `ci_sanity.sh` and build Tensorflow correctly.

We need this (python3.8) to let the user run `ci_sanity.sh` on a reference developer local env.

**Comment by bhack** [comment]
Do you need anything else here?

**Comment by bhack** [comment]
Gently ping

**Comment by vfrico** [comment]
Having Python 3.8 on official docker images would be a fantastic feature. Is there anything I can help?

**Comment by bhack** [comment]
Weekly ping

**Comment by gmabey** [comment]
@angerson gently weakly 
1f3d3

**Comment by gbaned** [comment]
@angerson Can you please review this PR ? Thanks!

**Comment by bhack** [comment]
We passed 4 months. 
I think that currently we don't have the resource to review this, thanks anyway

**Pull request closed without merging**